sql/planner/projection.go

```diff
@@ -37,6 +38,18 @@ func NewProjectionNode(n Node, expressions []ProjectedField, tableName string) N
 	}
 }
 
+func NewDistinctProjectionNode(n Node, expressions []ProjectedField, tableName string) Node {
```

Comment:
Distinct should be its own node, all it does is filtering duplicate documents with the same shape.
We could add a `Dedup` operation, that would deduplicate any document from a stream, using Filter with a document Hashset.

Here is how it would look like and where it would be inserted in the final stream:

```sql
genji> EXPLAIN SELECT a, b FROM foo WHERE c > 10 ORDER BY d LIMIT 1 OFFSET 2;
{
  "plan": "Table(foo) -> σ(cond: c > 10) -> ∏(a, b) -> Sort(d ASC) -> Offset(2) -> Limit(1)"
}

genji> EXPLAIN SELECT DISTINCT a, b FROM foo WHERE c > 10 ORDER BY d LIMIT 1 OFFSET 2;
{
  "plan": "Table(foo) -> σ(cond: c > 10) -> ∏(a, b) -> Dedup() -> Sort(d ASC) -> Offset(2) -> Limit(1)"
}
```